net-dns/yaddns/yaddns-1.1.recipe

```diff
@@ -0,0 +1,70 @@
+SUMMARY="A small dynamic DNS client"
+DESCRIPTION="Yaddns is a small dynamic dns client with multiple servers \
+support and a high flexibility."
+HOMEPAGE="https://yaddns.github.io/"
+COPYRIGHT="2015 Anthony Viallard and Raphael Huck"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/yaddns/yaddns/archive/yaddns-$portVersion.tar.gz"
+CHECKSUM_SHA256="23b527b7f71e1449746f1e39cbc3a074d58bd261a33399fb4e2d90eb63149059"
+SOURCE_DIR="yaddns-yaddns-$portVersion"
+PATCHES="yaddns-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+commandBinDir=$binDir
+commandSuffix=$secondaryArchSuffix
+if [ "$targetArchitecture" = x86_gcc2 ]
+then
+	commandBinDir=$prefix/bin
+	commandSuffix=
+fi
+
+PROVIDES="
+	yaddns$secondaryArchSuffix = $portVersion
+	cmd:yaddns$commandSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:aclocal
+	cmd:autoconf
+	cmd:autom4te
+	cmd:automake
+	cmd:autoreconf
+	cmd:g++$secondaryArchSuffix
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	cmd:makeinfo
+	"
+
+USER_SETTINGS_FILES="
+	settings/yaddns directory
```

Comment:
Shouldn't there be an

    settings/yaddns/yaddns.conf template $dataDir/yaddns/yaddns.conf.example

entry?

Reply:
If the example config was OK as is, then I would also recommend adding the template part.
The problem is, if I'm not mistaken, that the example config has settings which, if not modified, would make the program contact a server and try to do a dyndns update... with a fake domain and fake user credentials.

We could keep the example config unchanged, and install a modified config, but in that case I think it would not be OK to use the `template` keyword. (If we install a config which has a template, then I think it has to be identical to the template, but I might be excessively strict.)

Another solution would be to change the example config by commenting everything which would cause yaddns to actually contact a server if the user does not first do the customization of his config.
The downside of this is that we can't call this an example config anymore, since yaddns would complain. But maybe this is acceptable.

The last solution, could be to not use the `template` keyword (and not install a default config), but, instead, patch the man page to let the user know that there is an example config in `/boot/system/data/yaddns/yaddns.conf.example`

Of course, the patch could to be made in the `PATCH()` function and use a
```
"`finddir B_SYSTEM_DATA_DIRECTORY`/yaddns/yaddns.conf.example"
```
in a `sed` command.

Reply:
@fbrosson thanks for the reminder, I've looked into it

The example config is horrible, so I would recommend not having it at all. The user could construct an better config with the man page anyway.

Reply:
Yep, let's drop the example config. The man page (and Google) should be enough :)
@sfanxiang, could you please remove everything in `INSTALL()` and only leave the calle to "make install"?
(There's also the `PATCH()` function to replace `/etc/yaddns.conf` by `/boot/system/settings/yaddns.conf` in `yaddns.1` and the small change in the patchset.)
Thanks!

Reply:
@fbrosson I try to follow your conversations and I'll update the patchset when I am on a computer.

If you all think removing the configuration file is a better choice then I don't object, but I can't help thinking that, as a user, I would actually prefer a demonstrative config file filled with pseudo values over those manpages (the manpages are important too, of course, but I'll go to the config first).

When I port a project, I attempt to replicate the author's intention as much as possible, but Haiku's philosophy more values adhering to elaborate rules. Perhaps the rules work better on a larger scale.

Reply:
OK. No problem. (There is no urgency at all, by the way.)

Regarding the example config, after all, we could install it in `$docDir` (which will appear in `/boot/system/documentation/packages/yaddns/`) . And while at it, we could also install the AUTHORS file in there too.
```diff
 INSTALL()
 {
 	make install
+	install -d $docDir
+	install AUTHORS $docDir
+	install etc/yaddns.conf $docDir/yaddns.conf.example
 }
```
@alaviss (and @sfanxiang), what do you think?